app/code/GorbanSv/AskQuestion/Observer/Model/ResourceModel/LoadBefore.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace GorbanSv\AskQuestion\Observer\Model\ResourceModel;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Registry;
+
+/**
+ * Class LoadBefore
+ * @package GorbanSv\AskQuestion\Observer\Model\ResourceModel
+ */
+class LoadBefore implements ObserverInterface
+{
+    /**
+     * @var Registry 
+     */
+    private $registry;
+
+    public function __construct(Registry $registry)
+    {
+        $this->registry = $registry;
+    }
+
+    /**
+     * @param Observer $observer
+     */
+    public function execute(Observer $observer)
+    {
+        $observer->getEvent()
+                 ->getData('ask_question_collection_object')
+                 ->addFieldToFilter('sku', $this->registry->registry('product')->getSku());
```

Comment:
This code could fail if there is no current product in the registry (if for some reason you are loading the questions collection not on the product page).
Please add the condition if $this->registry->registry('product') is not null and skip this operation if null